### ISSUE TITLE
Retrieving available authentication methods

### ISSUE THREAD

**Issue by tim-sim** [hierynomus/sshj#106]
Hello.
I've started to use your library recently. Everything works great and I very grateful to your work, but I have a modest request.

The class UserAuthImpl stores list of authentication methods supported by server in private variable allowedMethods and there is no way to get it from the outside. Also, it's populated by actual server response only after authentication failure (in handle() method). It would be nice to have a feature to retrieve list of supported authentication methods without login attempt. I mean method that looks like this:

List<String> getAllowedAuthMethods(String username)

Thanks,
Timur Nasibullin timur.nasibullin@gmail.com
